addon/components/variable-number/number.ts

```diff
@@ -44,7 +44,18 @@ export default class VariableNumberPluginNumberComponent extends Component<Args>
     return this.args.node;
   }
   get formattedNumber() {
-    return this.node.attrs.value as string;
+    const writtenNumber = Boolean(this.writtenNumber);
+
+    if (
+      !writtenNumber ||
+      Number.isNaN(Number(this.node.attrs.value)) ||
+      this.node.attrs.value === null ||
+      this.node.attrs.value === ''
+    ) {
+      return this.node.attrs.value as string;
+    } else {
+      return n2words(Number(this.node.attrs.value), { lang: 'nl' });
```

Comment:
This would only be needed if invalid html is imported (as normally attrs.value is always a valid number because of the validation). But as you validate the input with contentToDom, this isn't really needed.

Besides that, you'd want to split this if statement anyway, with an if/else using only `!writtenNumber`.